src/libraries/System.Private.Xml/src/System/Xml/Core/XsdValidatingReaderAsync.cs

```diff
@@ -506,13 +506,15 @@ private Task ProcessReaderEventAsync()
 
                 case XmlNodeType.Whitespace:
                 case XmlNodeType.SignificantWhitespace:
-                    _validator.ValidateWhitespace(GetStringValue);
-                    break;
+
+                    return GetValueAsync()
+                        .ContinueWith(t => _validator.ValidateWhitespace(t.GetAwaiter().GetResult()), TaskScheduler.Default);
```

Comment:
if 'GetValueAsync' returns `Task<string>' then you can just use t.Result here instead of going through the awaiter to get the result.

Reply:
If the operation failed, using Result will wrap the exception in an AggregateException.